Keen.NetStandard.Test/EventCacheTest.cs

```diff
@@ -0,0 +1,110 @@
+            await cache.Add(new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_AddNotEmpty_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            Assert.Null(await cache.TryTake());
+            await cache.Add(new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            Assert.NotNull(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_AddClearEmpty_Success(IEventCache cache)
+        {
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            await cache.Clear();
+            Assert.Null(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_Iterate_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            Assert.NotNull(await cache.TryTake());
+            Assert.NotNull(await cache.TryTake());
+            Assert.Null(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_SendEmptyEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+            Assert.DoesNotThrow(client.SendCachedEvents);
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_ClearEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+            Assert.DoesNotThrow(() => client.EventCache.Clear());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_AddEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task CachingPCL_SendEventsParallel_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            var client = new KeenClient(SettingsEnv, cache);
+            if (UseMocks)
+                client.Event = new EventMock(SettingsEnv,
+                                             addEvents: (e, p) => new List<CachedEvent>());
+
+            (from i in Enumerable.Range(1,100)
+             select new { Property = "Value" })
```

Comment:
Not sure about the indentation style guidelines here. We should choose how that'll work, same with the `EventMock` ctor params below. We should settle on some style guidelines and actually formulate a settings file or StyleCop rules or something. Do whatever looks fine for now and doesn't go over like 100 characters.